plugin/audit_log/audit_log.c

```diff
@@ -186,16 +187,29 @@ void escape_buf(const char *in, size_t *inlen, char *out, size_t *outlen,
   while ((in < inend) && (out < outend))
   {
     replaced= FALSE;
-    for (rule= escape_rules; rule->character; rule++)
-    {
-      if (*in == rule->character)
-      {
+    if ((unsigned char)(*in) < 32) {
+      if (control_escape_rules[(unsigned int)*in].character) {
+        rule = &control_escape_rules[(unsigned int)*in];
+
         if ((outend - out) < (int) rule->length)
           goto end_of_buffer;
         memcpy(out, rule->replacement, rule->length);
         out += rule->length;
         replaced= TRUE;
-        break;
+      }
+    } else
+    {
+      for (rule= other_escape_rules; rule->character; rule++)
+      {
+        if (*in == rule->character)
+        {
+          if ((outend - out) < (int) rule->length)
```

Comment:
Why a typecast here? If it has to stay, wouldn't it be more exact to use ptrdiff_t?